pkg/controller/controller.go

```diff
@@ -185,6 +189,17 @@ func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 		}
 	}
 
+	c.clusterIDLock.Lock()
```

Comment:
This still looks out of place. It should be a method with a comment block to understand why this is being done. This should not just be added to the Run method directly.